cpp/src/Slice/Parser.h

```diff
@@ -158,14 +162,61 @@ namespace Slice
         virtual bool shouldVisitIncludedDefinitions() const { return false; }
     };
 
+    // ----------------------------------------------------------------------
+    // GrammarBase
+    // ----------------------------------------------------------------------
+
+    class GrammarBase
+    {
+    public:
+        virtual ~GrammarBase() = default;
+    };
```

Comment:
This was just moved up from lower in the file. This definition has to come before `Metadata`, and `Metadata` has to come before some other things which also use it.